src/main/java/bookstore/service/OrderServiceImpl.java

```diff
@@ -0,0 +1,120 @@
+package bookstore.service;
+
+import bookstore.dto.order.OrderCreateRequestDto;
+import bookstore.dto.order.OrderDto;
+import bookstore.dto.order.OrderItemDto;
+import bookstore.dto.order.OrderStatusPatchDto;
+import bookstore.entity.CartItem;
+import bookstore.entity.Order;
+import bookstore.entity.OrderItem;
+import bookstore.entity.User;
+import bookstore.exception.EntityNotFoundException;
+import bookstore.mapper.OrderItemMapper;
+import bookstore.mapper.OrderMapper;
+import bookstore.repository.CartItemRepository;
+import bookstore.repository.OrderItemRepository;
+import bookstore.repository.OrderRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+@Transactional
+public class OrderServiceImpl implements OrderService {
+
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final CartItemRepository cartItemRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+
+    @Override
+    public Page<OrderDto> getOrders(Authentication authentication,
+                                    Pageable pageable) {
+        User user = (User) authentication.getPrincipal();
+        return orderRepository.findByUserId(user.getId(), pageable)
+                .map(orderMapper::toDto);
+    }
+
+    @Override
+    public OrderDto createOrder(Authentication authentication,
+                                OrderCreateRequestDto requestDto) {
+        User user = (User) authentication.getPrincipal();
+
+        Order newOrder = orderMapper.toModelForCreating(requestDto);
+        newOrder.setUser(user);
+        newOrder.setStatus(Order.Status.PENDING);
+        newOrder.setOrderDate(LocalDateTime.now());
+
+        Set<CartItem> cartItems = cartItemRepository.findByCartUserId(user.getId());
+
+        BigDecimal total = cartItems.stream()
+                .map(cartItem -> cartItem.getBook().getPrice()
+                        .multiply(BigDecimal.valueOf(cartItem.getQuantity())))
+                .reduce(BigDecimal.ZERO, BigDecimal::add);
+        newOrder.setTotal(total);
+
+        newOrder = orderRepository.saveAndFlush(newOrder);

```

Comment:
Why saveAndFlush?

Reply:
saveAndFlush() is used to save the entity and actually commit the changes to the database, ensuring data persistence before further operations. This is important when you need the changes to be reflected in the database immediately, such as when calculating the total cost of an order after saving.

Reply:
Does it work with the regular save method? I guess so